receiver/datadogreceiver/receiver_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadogreceiver
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/obsreport"
+	"go.opentelemetry.io/collector/receiver"
+)
+
+func TestDatadogReceiver_Lifecycle(t *testing.T) {
+
+	ddr, err := newDataDogReceiver(&Config{
+		ReceiverSettings: obsreport.ReceiverSettings{ReceiverID: component.ID{}},
+		HTTPServerSettings: confighttp.HTTPServerSettings{
+			Endpoint: "localhost:8126",
+		},
+	}, consumertest.NewNop(), receiver.CreateSettings{})
+
+	assert.NoError(t, err, "Receiver should be created")
+
+	err = ddr.Start(context.Background(), componenttest.NewNopHost())
+	assert.NoError(t, err, "Server should start")
+
+	err = ddr.Shutdown(context.Background())
+	assert.NoError(t, err, "Server should stop")
```

Comment:
If you repeat this sequence of event (start/shurdown) a couple time, this would trigger the deadlock that is being feared by reviewers.

Reply:
This was resolved by the refactored code in  the Start() function